lib/spack/spack/test/llnl/util/lock.py

```diff
@@ -670,15 +689,18 @@ def test_upgrade_read_to_write_fails_with_readonly_file(private_lock_path):
         lock = lk.Lock(private_lock_path)
         assert lock._reads == 0
         assert lock._writes == 0
+        assert lock._current_lock is None
 
         lock.acquire_read()
         assert lock._reads == 1
         assert lock._writes == 0
         assert lock._file.mode == 'r'
+        assert lock._current_lock == lock.LOCK_SH
```

Comment:
Would it make sense to move these out along with the lock changes, or are these patches required even though these tests no longer run on Windows?

Reply:
Yes, absolutely, I thought I had spun these out with the lock changes but they must have been lumped in with the other testing changes. I'll revert the change here.

Reply:
Note: I will do this closer to the date of merge, as it will involve some rebasing which has led to reviewing difficulties thus far.